src/ray/common/ray_config_def.h

```diff
@@ -398,7 +398,9 @@ RAY_CONFIG(bool, enable_metrics_collection, true)
 RAY_CONFIG(int64_t, task_rpc_inlined_bytes_limit, 10 * 1024 * 1024)
 
 /// Maximum number of pending lease requests per scheduling category
-RAY_CONFIG(uint64_t, max_pending_lease_requests_per_scheduling_category, 10)
+/// -1 means that Ray should automatically set this to the number of nodes in
```

Comment:
Why is this number of nodes instead of number of cpus? In that case, we could solve the problem independent of scheduling policy and as an added bonus, not have to rely on worker pre-starting too.

Reply:
This is to try to make sure that each node gets at least one request, but we can still take advantage of lease reuse for tasks that are short.

Not sure what you mean by "independent of scheduling policy"?

Reply:
I mean that if parallelism = num_cpus_in_cluster, then we would try our best to parallelize across the entire cluster regardless of what the hybrid scheduling threshold is.

Reply:
Actually even with what @wuisawesome suggests, we still have lease reuse. We may just have some wasted leases?